configuration/pih/liquibase/sql/void_duplicate_registrations.sql

```diff
@@ -0,0 +1,158 @@
+-- variables 
+select encounter_type_id  into @regEncounterType from encounter_type et where uuid='873f968a-73a8-4f9c-ac78-9f4778b751b6';
+
+-- create temp table to load
+DROP TEMPORARY TABLE IF EXISTS temp_patients;
+CREATE TEMPORARY TABLE  temp_patients
+(patient_id                        int,           
+mothers_first_name                VARCHAR(255),  
+registration_encounter_id         int(11),       
+address_level_1                   VARCHAR(255),  
+address_level_4                          VARCHAR(255),  
+address_level_2                          VARCHAR(255),  
+address_level_5                           VARCHAR(255),  
+address_level_3                           VARCHAR(255),  
+telephone_number                  VARCHAR(255),  
+reg_location                      varchar(50),   
+reg_location_id                   int(11),       
+registration_date                 date,          
+registration_entry_date           datetime,      
+creator                           int(11),       
+user_entered                      varchar(50),   
+first_encounter_date              date,          
+last_encounter_date               date,          
+name                              varchar(50),   
+family_name                       varchar(50),   
+dob                               date,          
+gender                            varchar(2)    
+);
+
+-- load all patients
+insert into temp_patients (patient_id) 
+select patient_id from patient p where p.voided = 0;
+create index temp_patients_pi on temp_patients(patient_id);
+
+-- person info
+update temp_patients t
+inner join person p on p.person_id = t.patient_id
+set t.gender = p.gender,
+	t.dob = p.birthdate;
+
+-- name info
+update temp_patients t
+inner join person_name n on n.person_name_id =
+	(select n2.person_name_id from person_name n2
+	where n2.person_id = t.patient_id
+	order by preferred desc, date_created desc limit 1)
+set t.name = n.given_name,
+	t.family_name = n.family_name;
+
+-- address info
+update temp_patients t
+inner join person_address a on a.person_address_id =
+	(select a2.person_address_id from person_address a2
+	where a2.person_id = t.patient_id
+	order by preferred desc, date_created desc limit 1)
+set t.address_level_1 = a.country,
+	t.address_level_2 = a.state_province,
+	t.address_level_3 = a.city_village,
+	t.address_level_4 = a.county_district ,
+	t.address_level_5 = a.address1;
+
+-- person attributes
+select person_attribute_type_id into @telephone from person_attribute_type where name = 'Telephone Number' ;
+select person_attribute_type_id into @motherName from person_attribute_type where name = 'First Name of Mother' ;
```

Comment:
Not a big deal, but do we connect these attributes in all our countries?  I guess if not just needs to be null-safe.

Reply:
Right - I made sure that things were null-safe.